plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/animator/CameraAnimator.kt

```diff
@@ -42,8 +44,8 @@ abstract class CameraAnimator<out T> (
   private var internalUpdateListener: AnimatorUpdateListener? = null
   private var internalListener: AnimatorListener? = null
 
-  private val userUpdateListeners = mutableListOf<AnimatorUpdateListener?>()
-  private val userListeners = mutableListOf<AnimatorListener?>()
+  private val userUpdateListeners = CopyOnWriteArraySet<AnimatorUpdateListener?>()
+  private val userListeners = CopyOnWriteArraySet<AnimatorListener?>()
```

Comment:
why are those copy on write sets? first issue I see - why is it a set, imo adding duplicates should be allowed; another thing - we is it copy on write? we indeed need to operate on local copy of `listeners` below but are those needed to be copy on write?

Reply:
I've also started the conv in above comment [here](https://github.com/mapbox/mapbox-maps-android/pull/1842#discussion_r1023831004)

Reply:
discussed offline, seems reasonable